pkg/render/logstorage.go

```diff
@@ -572,7 +568,7 @@ func (es elasticsearchComponent) podTemplate() corev1.PodTemplateSpec {
 			SecurityContext: &corev1.SecurityContext{
 				Privileged: ptr.BoolToPtr(false),
 			},
-			Args: []string{"bash", "-c", "/mnt/elastic-internal/scripts/prepare-fs.sh"},
+			Command: []string{"bash", "-c", "mkdir /mnt/elastic-internal/transport-certificates/ && touch /mnt/elastic-internal/transport-certificates/$HOSTNAME.tls.key && /mnt/elastic-internal/scripts/prepare-fs.sh"},
```

Comment:
Not sure why we need to create the `transport-certificates` folder and the empty `$HOSTNAME.tls.key` file but the volume mount is removed below?

Reply:
The prepare-fs script waits until a file if present in this location. Normally, it would use that secret and place it in the location that the elasticsearch.yml file uses when booting elastic. However, we don't want an ECK secret, we want the CSR init container to create a certificate request and use that instead. This `touch` command lets the `prepare-fs` script finish.